### PR TITLE
[REVIEW] Fix slice issue with empty column

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
When an empty column is being sliced with start and end {0, 0}, normally we would expect an empty column of same type, but it was returning empty vector.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/tests/copying/slice_tests.cpp [hunk 1]
```diff
@@ -262,11 +262,11 @@ struct SliceCornerCases : public SliceTest<int8_t> {
 TEST_F(SliceCornerCases, EmptyColumn)
 {
   cudf::column col{};
-  std::vector<cudf::size_type> indices{1, 3, 2, 4, 5, 9};
+  std::vector<cudf::size_type> indices{0, 0, 0, 0, 0, 0};
 
   std::vector<cudf::column_view> result = cudf::slice(col.view(), indices);
 
-  unsigned long expected = 0;
+  unsigned long expected = 3;
```
Comment (jrhemstad) [inline cpp/tests/copying/slice_tests.cpp:269]: Should probably check the types too?

File: cpp/src/copying/slice.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ std::vector<column_view> slice(column_view const& input,
 
   std::vector<column_view> result{};
 
-  if (indices.size() == 0 or input.size() == 0) { return result; }
+  if (indices.size() == 0) { return result; }
```
Comment (codereport) [inline cpp/src/copying/slice.cpp:36]: Recommend using `empty` and moving `result` towards the bottom and just return `{}` here.
```suggestion
  if (indices.empty()) return {};
```

File: cpp/src/copying/slice.cpp [hunk 3]
```diff
@@ -33,7 +33,7 @@ std::vector<column_view> slice(column_view const& input,
 
   std::vector<column_view> result{};
```
Comment (codereport) [inline cpp/src/copying/slice.cpp:34]: Move `std::vector<column_view> result{};` to line 42 above `for` loop.
```suggestion
```

**Review by codereport — CHANGES_REQUESTED** [review]

File: cpp/src/copying/slice.cpp [hunk 4]
```diff
@@ -72,9 +71,7 @@ std::vector<cudf::table_view> slice(cudf::table_view const& input,
 {
   CUDF_FUNC_RANGE();
   CUDF_EXPECTS(indices.size() % 2 == 0, "indices size must be even");
-  std::vector<cudf::table_view> result{};
-
-  if (indices.size() == 0 or input.num_columns() == 0) { return result; }
```
Comment (codereport) [inline cpp/src/copying/slice.cpp:77]: I assume that you dropped the `input.num_columns() == 0` intentionally and checking `empty()` on `indices` is equivalent?

**Review by codereport — APPROVED** [review]

File: cpp/src/copying/slice.cpp (hunk 4 above)
Comment (rgsl888prabhu) [reply cpp/src/copying/slice.cpp:77]: no, both were two different conditions. I added this condition while creating this API (Not sure why, I can't track back to any request and design), and it is normal to expect number of columns as number of slices requested. So, removed this change.

Comment (harrism) [reply cpp/src/copying/slice.cpp:77]: I don't think the number of slices should match the number of columns. I think the number of slices is the number of output tables -- all output tables will have a slice across all columns.

**Review by harrism — APPROVED** [review]

File: cpp/src/copying/slice.cpp (hunk 4 above)
Comment (rgsl888prabhu) [reply cpp/src/copying/slice.cpp:77]: I meant in case of column is sliced, in case of table as you mentioned set of tables.

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**